server/server.go

```diff
@@ -105,6 +105,7 @@ import (
 
 const maxConcurrentLoginRequestsCountEnv = "ARGOCD_MAX_CONCURRENT_LOGIN_REQUESTS_COUNT"
 const replicasCountEnv = "ARGOCD_API_SERVER_REPLICAS"
+const renewTokenKey = "renew-token"
```

Comment:
Can you explain how this works? Is this is a new cookie key and why it needed in addition to the argocd-token?

Reply:
We need to somehow return regenerated token from the grpc service back to grpc gateway so it can be translated into the cookie.

GRPC provides outgoing server metadata for such purposes but GRPC gateway generates code that allows accessing only `grpc-gateway.ServerMetadata` that contains "headers" and "trailers" metadata. 

So "Authenticate" method adds regenerated token into  outgoing `ServerMetadata.HeaderMD`  field. Then grpc-gateway response forwarder ( `translateGrpcCookieHeader` ) transforms this header into a proper cookie header

Reply:
Ah, I see. So it's not a key that a client is ever exposed to and is only used for internal purposes due to the way grpc-gateway works. 

Do you mind adding a comment on how this works somewhere in the code for future reference?

Reply:
Sure. adding comment about it